src/actions/getAllDeposits.ts

```diff
@@ -14,18 +14,17 @@ export const getAllDeposits = async (
 
   if (depositLength.eq(ethers.BigNumber.from("0"))) return [];
 
-  const deposits: Deposit[] = [];
-
+  const promiseArray: Promise<any>[] = [];
+  
   for (let i = 0; i < depositLength.toNumber(); i++) {
-    const deposit = await corePool.getDeposit(
+    const depositPromise = corePool.getDeposit(
       address,
       ethers.BigNumber.from(i)
     );
-    // Deposits are left in array even after being unstaked
```

Comment:
This comment should stay, it explains the "why" of this filtering logic which are the most helpful types of comment.